# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: xpcom/base/CycleCollectedJSRuntime.cpp
# Commit: ebbd2cd3905e
# Full Hash: ebbd2cd3905e3542c417ea33e0c2cdd88260e6c9
# Author: Cristina Horotan <chorotan@mozilla.com>
# Date: 2025-05-22 09:08:48
# Regressor Bug: 1964451
# File Overlap Count: 1
# Description:
#   Revert "Bug 1964451 - Add use counters for OOM handling r=sfink,mccr8" on request
#   
#   This reverts commit 7ae56a029252266ce967356e2254b1408b8522ea.
# ==============================================================================

diff -r 34bb3734b886 -r ebbd2cd3905e xpcom/base/CycleCollectedJSRuntime.cpp
--- a/xpcom/base/CycleCollectedJSRuntime.cpp	Wed May 21 18:52:10 2025 +0000
+++ b/xpcom/base/CycleCollectedJSRuntime.cpp	Wed May 21 22:38:12 2025 +0300
@@ -1992,14 +1992,7 @@
 
 void CycleCollectedJSRuntime::AnnotateAndSetOutOfMemory(OOMState* aStatePtr,
                                                         OOMState aNewState) {
-  enum class Size { Large, Small };
-
-  Size size = aStatePtr == &mOutOfMemoryState ? Size::Small : Size::Large;
-  MOZ_ASSERT_IF(size == Size::Large,
-                aStatePtr == &mLargeAllocationFailureState);
-
   *aStatePtr = aNewState;
-
   CrashReporter::Annotation annotation =
       (aStatePtr == &mOutOfMemoryState)
           ? CrashReporter::Annotation::JSOutOfMemory
@@ -2007,29 +2000,6 @@
 
   CrashReporter::RecordAnnotationCString(annotation,
                                          OOMStateToString(aNewState));
-
-  // Attempt to report telemetry.
-  if (JSObject* global = JS::CurrentGlobalOrNull(GetContext()->Context())) {
-    if (aNewState == OOMState::Recovered) {
-      switch (size) {
-        case Size::Large:
-          SetUseCounter(global, eUseCounter_custom_JS_large_oom_recovered);
-          break;
-        case Size::Small:
-          SetUseCounter(global, eUseCounter_custom_JS_small_oom_recovered);
-          break;
-      }
-    } else {
-      switch (size) {
-        case Size::Large:
-          SetUseCounter(global, eUseCounter_custom_JS_large_oom_reported);
-          break;
-        case Size::Small:
-          SetUseCounter(global, eUseCounter_custom_JS_small_oom_reported);
-          break;
-      }
-    }
-  }
 }
 
 void CycleCollectedJSRuntime::OnGC(JSContext* aContext, JSGCStatus aStatus,
